src/modules/authorizer/IAuthorizer.sol

```diff
@@ -1,8 +1,90 @@
+
+    //--------------------------------------------------------------------------
+    // Functions
+
+    /// @notice Asks whether an address holds the required module role to execute
+    ///         the current transaction.
+    /// @param role The identifier of the role we want to check
+    /// @param who  The address on which to perform the check.
+    /// @dev It will use the calling address to generate the role ID. Therefore, for checking on anything other than itself, hasRole() should be used
+    function hasModuleRole(bytes32 role, address who)
+        external
+        view
+        returns (bool);
+
+    /// @notice Helper function to generate a bytes32 role hash for a module role
+    /// @param module The address of the module to generate the hash for
+    /// @param role  The ID number of the role to generate the hash for
+    function generateRoleId(address module, bytes32 role)
+        external
+        returns (bytes32);
+
+    /// @notice Used by a Module to grant a role to a user.
+    /// @param role The identifier of the role to grant
+    /// @param target  The address to which to grant the role.
+    function grantRoleFromModule(bytes32 role, address target) external;
+
+    /// @notice Used by a Module to revoke a role from a user.
+    /// @param role The identifier of the role to revoke
+    /// @param target  The address to revoke the role from.
+    function revokeRoleFromModule(bytes32 role, address target) external;
+
+    /*     /// @notice Toggles if a Module self-manages its roles or defaults to the orchestrator's roles.
+    function toggleModuleSelfManagement() external; */
+
+    /// @notice Transfer the admin rights to a given role.
+    /// @param roleId The role on which to peform the admin transfer
+    /// @param newAdmin The new role to which to transfer admin access to
+    function transferAdminRole(bytes32 roleId, bytes32 newAdmin) external;
+
+    /// @notice Irreversibly burns the admin of a given role.
+    /// @param role The role to remove admin access from
+    /// @dev The module itself can still grant and revoke it's own roles. This only burns third-party access to the role.
+    function burnAdminFromModuleRole(bytes32 role) external;
+
+    /// @notice Grants a global role to a target
+    /// @param role The role to grant
+    /// @param target The address to grant the role to
+    /// @dev Only the addresses with the Owner role should be able to call this function
+    function grantGlobalRole(bytes32 role, address target) external;
+
+    /// @notice Revokes a global role from a target
+    /// @param role The role to grant
+    /// @param target The address to grant the role to
+    /// @dev Only the addresses with the Owner role should be able to call this function
+    function revokeGlobalRole(bytes32 role, address target) external;
+
+    /// @notice Returns the role ID of the owner role
+    /// @return The role ID
+    function getOwnerRole() external view returns (bytes32);
```

Comment:
The idea was to have something in the interface which modules can refer to. Do you think this is a bad idea?

Reply:
No, I do not think that is a bad idea at all in itself, but unfortunately solidity does not allow for constants to be part of the interface definition, so you cannot do 

```
interface IAuthorizer {
     // this will not compile
     bytes32 public constant OWNER_ROLE = 0x3231333433323433320000000000000000000000000000000000000000000000; // keccak256("OWNER_ROLE")
}


```
What you can do instead is this

```
// IAuthorizer.sol
bytes32 constant OWNER_ROLE = 0x3231333433323433320000000000000000000000000000000000000000000000; // keccak256("OWNER_ROLE")

interface IAuthorizer {
   // define functions sigs like hasRole    
}
```

and then use it like this:

```
// Consumer.sol
import "IAuthorizer.sol";

contract Foo {
    function functionThatUsesOwnerRole() public {
        bytes32 foo = OWNER_ROLE;
        IAuthorizer(authorizerAddress).hasRole(myAddress, OWNER_ROLE);
    }
}
```

Reply:
Discarded for now, see https://github.com/InverterNetwork/inverter-contracts/issues/370#issuecomment-1792694618